File: gaia-sdk-javascript/src/graphql/request/type/Intent.ts


import VariableRegistry from "../../../api/VariableRegistry"
import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {RuntimeState} from "../enumeration/RuntimeState";
import {SkillState} from "../enumeration/SkillState";

/**
 * this type represents the intent information
 */
export class Intent extends Array<(_:VariableRegistry) => string> {

    /**
     * The identity id
     */
    public identityId = () => { 
        this.push(_ => "identityId")
    };

    /**
     * The intent reference id
     */
    public reference = () => { 
        this.push(_ => "reference")
    };

    /**
     * The name of the intent
     */
    public qualifier = () => { 
        this.push(_ => "qualifier")
    };

    /**
     * Detailed description about the intent
     */
    public appendent = () => { 
        this.push(_ => "appendent")
    };

    /**
     * The utterance dictionary. The key is a language key and the value is a list of utterances
     */
    public utterance = () => { 
        this.push(_ => "utterance")
    };

    /**
     * The list of labels of the intent
     */
    public labelList = () => { 
        this.push(_ => "labelList")
    };

    /**
     * The version of the intent
     */
    public version = () => { 
        this.push(_ => "version")
    };

    public render = (registry: VariableRegistry):String => this.map(e => e(registry)).join(" ");
}

File: gaia-sdk-java/gaia-sdk-graphql/src/main/kotlin/gaia/sdk/response/type/Intent.kt
package gaia.sdk.response.type

import gaia.sdk.Uuid
import gaia.sdk.ISO8601
import gaia.sdk.Struct
import gaia.sdk.response.intf.*
import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.annotation.JsonCreator
import gaia.sdk.request.enumeration.*

/**
* Represents intent information
*/
data class Intent @JsonCreator constructor(
    /**
    * The identity id
    */
    @JsonProperty("identityId") val identityId:Uuid? = null, 
    /**
    * The intent reference id
    */
    @JsonProperty("reference") val reference:Uuid? = null, 
    /**
    * The name of the intent
    */
    @JsonProperty("qualifier") val qualifier:String? = null, 
    /**
    * Detailed description about the intent
    */
    @JsonProperty("appendent") val appendent:String? = null, 
    /**
    * The utterance dictionary. The key is a language key and the value is a list of utterances
    */
    @JsonProperty("utterance") val utterance:Struct? = null, 
    /**
    * The list of labels of the intent
    */
    @JsonProperty("labelList") val labelList:List<String>? = null, 
    /**
    * The version of the intent
    */
    @JsonProperty("version") val version:String? = null
)